cs/src/core/Index/Common/Contexts.cs

```diff
@@ -12,12 +12,20 @@
 
 namespace FASTER.core
 {
-    internal enum OperationType
+    /// <summary>
+    /// The type of FASTER operation being done
+    /// </summary>
+    public enum OperationType
     {
+        /// <summary>Read operation</summary>
         READ,
+        /// <summary>Read-Modify-Write operation</summary>
         RMW,
+        /// <summary>Upsert operation</summary>
         UPSERT,
+        /// <summary>Insert operation (either Upsert or RMW resulted in a new record)</summary>
```

Comment:
Is INSERT also used in internal use of `OperationType`. Why not keep a separate enum for the user facing API compared to internal?